### ISSUE TITLE
Validation progress bar with metrics

### ISSUE THREAD

**Issue by ethanjperez** [PyTorchLightning/pytorch-lightning#1239]
## 🚀 Feature
Logging validation metrics throughout the duration of validation (e.g., current batch loss/acc or average batch loss/acc so far).

### Motivation

If the validation set is large, it'd helpp to know right away if I've loaded the wrong checkpoint or loaded a checkpoint in the wrong way within a couple iterations, rather than waiting to run through the entire validation set.

### Pitch

I want to be able to see basically the same progress bar during training as I do during validation/testing.

### Alternatives

Not sure of any alternatives - this is quite helpful for debugging and is probably a not-too-large change since training already supports this feature.

**Comment by williamFalcon** [comment]
in summary, we need to support:

```
validation_step(...):
  return {'log':{...}}
```

**Comment by Borda** [comment]
@ethanjperez mind submit a PR?

**Comment by williamFalcon** [comment]
@PyTorchLightning/core-contributors anyone?

**Comment by Borda** [comment]
will it it be also solved in #1989?
